modules/json/json-parser.c

```diff
@@ -251,6 +251,33 @@ json_parser_process_object(struct json_object *jso,
   }
 }
 
+static void
+json_parser_process_array(struct json_object *jso,
+                          const gchar *prefix,
+                          LogMessage *msg)
+{
+  gint i;
+
+  msg->num_matches = 0;
+  log_msg_unset_match(msg, 0);
+  for (i = 0; i < json_object_array_length(jso) && i < LOGMSG_MAX_MATCHES; i++)
+    {
+      struct json_object *el = json_object_array_get_idx(jso, i);
+      GString *element_value = scratch_buffers_alloc();
+      LogMessageValueType element_type;
+
+      if (json_parser_extract_string_from_simple_json_object(el, element_value, &element_type))
+        {
+          log_msg_set_match_with_type(msg, i + 1, element_value->str, element_value->len, element_type);
+        }
+      else
+        {
+          /* unknown type, encode the entire value as JSON */
+          log_msg_set_match_with_type(msg, i + 1, json_object_to_json_string_ext(el, JSON_C_TO_STRING_PLAIN), -1, LM_VT_JSON);
```

Comment:
Do we want to start the index from 1 instead of 0?

Reply:
I wouldn't $0 is always the entire string when doing regexps. With bash arguments $0 is the command itself and $1 is the first argument. $* does not contain $0.